version: "3.2"

services:
  adminer:
    container_name: adminer
    image: adminer:latest
    restart: always
    ports:
      - 8080:8080
    networks:
      - app-net
  wager:
    container_name: wager
    build:
      context: ${HUNCHO_DIR}/services/wager
      dockerfile: ${HUNCHO_DIR}/services/wager/build/Dockerfile
    restart: on-failure
    volumes:
      - ${HUNCHO_DIR}/services/wager/migrations:/home/app/migrations
      - ${HUNCHO_DIR}/services/wager/bin:/home/app/bin
      - ${HUNCHO_DIR}/services/wager/tests:/home/app/tests
      - ${HUNCHO_DIR}/services/wager/seeds:/home/app/seeds
      - ${HUNCHO_DIR}/services/wager/src:/home/app/src
      - ${HUNCHO_DIR}/services/wager/statics:/home/app/statics
      - ${HUNCHO_DIR}/services/wager/manage.py:/home/app/manage.py
      - ${HUNCHO_DIR}/services/wager/setup.py:/home/app/setup.py
    ports:
      - 20002:5000
    env_file:
      - ${HUNCHO_DIR}/env/dev/.env.wager
#    healthcheck:
#      test: curl --fail -s wager:5000/ping || exit 1
#      interval: 60s
#      timeout: 5s
#      retries: 5
    depends_on:
      - wager_db
    networks:
      - app-net
  wager_db:
    container_name: wager_db
    image: postgres:9.6
    volumes:
      - ${HUNCHO_DIR}/data/dev/wager_db/data:/var/lib/postgresql/data/
    env_file:
      - ${HUNCHO_DIR}/env/dev/.env.wager_db
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U wager"]
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - app-net
networks:
  app-net:
    driver: bridge